server/grpc_service.go

```diff
@@ -243,6 +244,13 @@ func (s *Server) RegionHeartbeat(server pdpb.PD_RegionHeartbeatServer) error {
 		if err != nil {
 			return errors.Trace(err)
 		}
+		if isNewStream {
+			storeID := request.GetLeader().GetStoreId()
+			s.streams.Lock()
+			s.streams.regionHeartbeatStreams[storeID] = server
+			s.streams.Unlock()
+			isNewStream = false
+		}
 
```

Comment:
it is not safe to send stream concurrently, the streams Lock can't guarantee the safety here. You should use a goroutine to send individually.